src/statement-list/converters/expr2statement_list.cpp

```diff
@@ -25,3 +117,187 @@ std::string type2stl(const typet &type, const namespacet &ns)
+    convert(to_not_expr(expr));
+  else // TODO: support more instructions in expr2stl.
+    return expr2c(expr, ns);
+
+  return result.str();
+}
+
+void expr2stlt::convert(const and_exprt &expr)
+{
+  std::vector<exprt> operands = expr.operands();
+  convert_multiary_bool(operands, AND);
+}
+
+void expr2stlt::convert(const or_exprt &expr)
+{
+  std::vector<exprt> operands = expr.operands();
+  convert_multiary_bool(operands, OR);
+}
+
+void expr2stlt::convert(const xor_exprt &expr)
+{
+  std::vector<exprt> operands = expr.operands();
+  convert_multiary_bool(operands, XOR);
+}
+
+void expr2stlt::convert(const notequal_exprt &expr)
+{
+  std::vector<exprt> operands = expr.operands();
+  convert_multiary_bool(operands, XOR);
+}
+
+void expr2stlt::convert(const equal_exprt &expr)
+{
+  std::vector<exprt> operands =
+    instrument_equal_operands(expr.lhs(), expr.rhs());
+  convert_multiary_bool(operands, XOR);
+}
+
+void expr2stlt::convert(const not_exprt &expr)
+{
+  // If a NOT expression is being handled here it must always mark the
+  // beginning of a new bit string.
+  PRECONDITION(!inside_bit_string);
+
+  if(ID_symbol == expr.op().id())
+  {
+    // Use AN to load the symbol.
+    is_reference = true;
+    result << AND << NOT_POSTFIX << OPERAND_SEPARATOR;
+    convert(to_symbol_expr(expr.op()));
+    result << LINE_SEPARATOR;
+  }
+  else
+  {
+    // Use NOT to negate the RLO after the wrapped expression was loaded.
+    convert(expr.op());
+    result << NOT LINE_SEPARATOR;
+  }
+}
+
+void expr2stlt::convert(const symbol_exprt &expr)
+{
+  if(is_reference)
+  {
+    result << REFERENCE_FLAG;
+    is_reference = false;
+  }
+  result << id2string(id_shorthand(expr.get_identifier()));
+}
+
+void expr2stlt::convert_multiary_bool(
+  std::vector<exprt> &operands,
+  const char operation)
+{
+  if(inside_bit_string)
+    convert_multiary_bool_operands(operands, operation);
+  else
+  {
+    convert_first_non_trivial_operand(operands);
+    convert_multiary_bool_operands(operands, operation);
+  }
+}
+
+void expr2stlt::convert_multiary_bool_operands(
+  const std::vector<exprt> &operands,
+  const char operation)
+{
+  for(const exprt &op : operands)
+  {
+    if(ID_not == op.id())
+    {
+      result << operation << NOT_POSTFIX;
+      convert_bool_operand(op.op0());
+    }
+    else
+    {
+      result << operation;
+      convert_bool_operand(op);
+    }
+  }
+}
+
+void expr2stlt::convert_bool_operand(const exprt &op)
+{
+  if(op.id() == ID_symbol)
+  {
+    is_reference = true;
+    result << OPERAND_SEPARATOR;
+    convert(op);
+    result << LINE_SEPARATOR;
+  }
+  else
+  {
+    inside_bit_string = false;
+    result << NESTING_OPEN_LINE_SEPARATOR;
+    convert(op);
+    result << NESTING_CLOSED_LINE_SEPARATOR;
+  }
+}
+
+void expr2stlt::convert_first_non_trivial_operand(std::vector<exprt> &operands)
+{
+  exprt non_trivial_op;
+  for(auto it(begin(operands)); it != end(operands); ++it)
```

Comment:
Use either constructor syntax `{...}` or equality operator `=`. The latter is for primitive types only, but an iterator is somewhere in between...